src/main/java/com/yahoo/memory/BaseWritableBufferImpl.java

```diff
@@ -59,45 +59,75 @@
   //DUPLICATES XXX
   @Override
   public Buffer duplicate() {
-    return writableDuplicateImpl(true);
+    return writableDuplicateImpl(true, getByteOrder());
   }
 
   @Override
   public WritableBuffer writableDuplicate() {
-    if (isReadOnly()) {
+    return writableDuplicateImpl(false, getByteOrder());
+  }
+
+  //Developer note: we don't allow switching byte order on duplicates.
+  //This is here to reduce complexity in the endian-sensitive classes and to allow us to easily
+  // change our mind in the future :)
+  WritableBuffer writableDuplicateImpl(final boolean localReadOnly, final ByteOrder byteOrder) {
+    if (isReadOnly() && !localReadOnly) {
       throw new ReadOnlyException("Writable duplicate of a read-only Buffer is not allowed.");
     }
-    return writableDuplicateImpl(false);
+    final WritableBuffer wbuf = Util.isNativeOrder(byteOrder)
+        ? new WritableBufferImpl(getUnsafeObject(), getNativeBaseOffset(),
+            getRegionOffset(), getCapacity(),
+            isReadOnly() || localReadOnly, getByteBuffer(), getValid(), originMemory)
```

Comment:
`isReadOnly() || localReadOnly` could be simplified to just `localReadOnly`. Same in other similar places